scripts/globals/homepoint.lua

```diff
@@ -1,361 +1,248 @@
+homepoints[104] = { 4,  9,     -65,     -5,      54, 127, 252,  500}; -- Norg #2                 
+homepoints[105] = { 4, 10,     -21,      8,     110,  64, 247,  500}; -- Rabao #2                
+homepoints[106] = { 4, 11,     130,      0,     -16,   0,  50,  500}; -- Aht Urhgan Whitegate #2 
+homepoints[107] = { 4, 12,    -108,     -6,     108, 192,  50,  500}; -- Aht Urhgan Whitegate #3 
+homepoints[108] = { 4, 13,     -99,      0,     -68,   0,  50,  500}; -- Aht Urhgan Whitegate #4 
+homepoints[109] = { 4, 14,      32,      0,    -164,  32, 256,  500}; -- Western Adoulin #2      
+homepoints[110] = { 4, 15,     -51,      0,     -96,  96, 257,  500}; -- Eastern Adoulin #2      
+homepoints[111] = { 4, 16,     223,    -13,    -254,   0, 137, 1000}; -- Xarcabard [S] #1        
+homepoints[112] = { 4, 17,       0,      0,       0,   0, 281, 1000}; -- Leafallia #1            
+homepoints[113] = { 4, 18,       0,      0,       0,   0, 155, 1000}; -- *Castle Zvahl Keep [S] #1
+
+local freeHpTeleGroups = { 1, 2, 3, 4, 5, 6, 7, 8};
+
+freeHpTeleGroups[1] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 97, 98}; --San d'Oria
+freeHpTeleGroups[2] = { 9, 10, 11, 12 ,13, 14, 15, 16, 99, 100, 101, 102}; -- Bastok
+freeHpTeleGroups[3] = { 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 ,27 ,28, 103}; -- Windurst
+freeHpTeleGroups[4] = { 29, 30, 31, 32, 33, 34, 35, 36, 37, 38}; -- Jueno
+freeHpTeleGroups[5] = { 65, 106, 107, 108}; -- Aht Urghan
+freeHpTeleGroups[6] = { 44, 45, 109, 110}; -- Adoulin
+freeHpTeleGroups[7] = { 41, 104}; -- Norg
+freeHpTeleGroups[8] = { 42, 105}; -- Rabao
+
+function homepointMenu( player, csid, hpIndex)
+	
+	if( HOMEPOINT_HEAL == 1) then
+		player:addHP(player:getMaxHP());
+		player:addMP(player:getMaxMP());
 	end
-	if(mask == 2) then
-		local hps = player:getVar("hpmask2")
-		if (bit.band(hps, bit.lshift(1, (num))) == 0)then
-			player:setVar("hpmask2",bit.bor(hps, bit.lshift(1, (num))))
-			return 65536
-		end
+
+	if( HOMEPOINT_TELEPORT == 1) then
+		
+		if( homepoints[hpIndex] == nil) then return; end -- Check for valid hpIndex
+
+		player:setLocalVar("currentHpIndex", hpIndex + 1);
+
+		local HpTeleportMask1 = bit.bor( bit.lshift( player:getVar("HpTeleportMask1a"), 16), player:getVar("HpTeleportMask1b"));
+		local HpTeleportMask2 = bit.bor( bit.lshift( player:getVar("HpTeleportMask2a"), 16), player:getVar("HpTeleportMask2b"));
+		local HpTeleportMask3 = bit.bor( bit.lshift( player:getVar("HpTeleportMask3a"), 16), player:getVar("HpTeleportMask3b"));
+		local HpTeleportMask4 = bit.bor( bit.lshift( player:getVar("HpTeleportMask4a"), 16), player:getVar("HpTeleportMask4b"));
+
+		-- Register new homepoint?
+		local newHp = 0;
+		if( homepoints[hpIndex][1] == 1) then
+			if( bit.rshift( bit.lshift( HpTeleportMask1, 32 - homepoints[hpIndex][2]), 31) == 0) then
+				newHp = 65536; -- This value causes the "You have registered a new home point!" dialog to display
```

Comment:
0x10000

obviously it's not a huge deal, but it's always a good idea to put the numbers in the format that is most readable - the numbers are never random!